src/actions.rs

```diff
@@ -0,0 +1,497 @@
+            PointerOrigin::WebElement(e) => WDActions::PointerOrigin::Element(
+                webdriver::common::WebElement(e.element_id().into()),
+            ),
+        }
+    }
+}
+
+impl PointerAction {
+    fn into_item(self) -> WDActions::PointerActionItem {
+        match self {
+            PointerAction::Pause { duration } => WDActions::PointerActionItem::General(
+                WDActions::GeneralAction::Pause(WDActions::PauseAction {
+                    duration: Some(duration.as_millis() as u64),
+                }),
+            ),
+            PointerAction::Down { button } => WDActions::PointerActionItem::Pointer(
+                WDActions::PointerAction::Down(WDActions::PointerDownAction { button }),
+            ),
+            PointerAction::Up { button } => WDActions::PointerActionItem::Pointer(
+                WDActions::PointerAction::Up(WDActions::PointerUpAction { button }),
+            ),
+            PointerAction::Move {
+                duration,
+                origin,
+                x,
+                y,
+            } => WDActions::PointerActionItem::Pointer(WDActions::PointerAction::Move(
+                WDActions::PointerMoveAction {
+                    duration: duration.map(|x| x.as_millis() as u64),
+                    origin: origin.into_wd_pointerorigin(),
+                    x: Some(x),
+                    y: Some(y),
+                },
+            )),
+            PointerAction::Cancel => {
+                WDActions::PointerActionItem::Pointer(WDActions::PointerAction::Cancel)
+            }
+        }
+    }
+}
+
+/// A sequence containing `Null` actions.
+#[derive(Debug, Clone)]
+pub struct NullActions {
+    /// An identifier to distinguish this sequence from others.
+    ///
+    /// Choose a meaningful string as it may be useful for debugging.
+    id: String,
+    /// The list of actions for this sequence.
+    actions: Vec<NullAction>,
+}
+
+impl NullActions {
+    /// Create a new NullActions sequence.
+    pub fn new(id: &str) -> Self {
+        Self {
+            id: id.to_string(),
+            actions: Vec::new(),
+        }
+    }
+}
+
+impl From<NullActions> for ActionSequence {
+    fn from(na: NullActions) -> Self {
+        ActionSequence(WDActions::ActionSequence {
+            id: na.id,
+            actions: WDActions::ActionsType::Null {
+                actions: na.actions.into_iter().map(|x| x.into_item()).collect(),
+            },
+        })
+    }
+}
+
+/// A sequence containing `Key` actions.
+#[derive(Debug, Clone)]
+pub struct KeyActions {
+    /// An identifier to distinguish this sequence from others.
+    ///
+    /// Choose a meaningful string as it may be useful for debugging.
+    id: String,
+    /// The list of actions for this sequence.
+    actions: Vec<KeyAction>,
+}
+
+impl KeyActions {
+    /// Create a new KeyActions sequence.
+    pub fn new(id: &str) -> Self {
+        Self {
+            id: id.to_string(),
```

Comment:
The convention is generally to let the caller handle allocation, so to take `String` rather than `&str` if we actually need a `String` internally.